scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/EcdhPskClientKeyExchange.java

```diff
@@ -0,0 +1,146 @@
+
+import org.eclipse.californium.elements.util.DatagramReader;
+import org.eclipse.californium.elements.util.DatagramWriter;
+import org.eclipse.californium.elements.util.StringUtil;
+import org.eclipse.californium.scandium.dtls.cipher.ECDHECryptography;
+import org.eclipse.californium.scandium.util.ByteArrayUtils;
+
+/**
+ * class EcdhPskClientKeyExchange. <br>
+ * This class EcdhPskClientKeyExchange represents the {@link ClientKeyExchange} in ECDHE_PSK mode. See <a
+ * href="https://tools.ietf.org/html/rfc5489#section-2">RFC 5489</a> for details.
+ * @author Vikram
+ *
+ */
+public final class EcdhPskClientKeyExchange extends ClientKeyExchange{
+
+
+	protected static final int LENGTH_BITS = 8; // opaque point <1..2^8-1>
+	private static final int IDENTITY_LENGTH_BITS = 16; // opaque <0..2^16-1>;
+
+	/**
+	 *See <a href="https://tools.ietf.org/html/rfc5489#section-2">RFC 5489</a>.
+	 */
+	private final byte[] identityEncoded;
+	/**
+	 *See <a href="https://tools.ietf.org/html/rfc5489#section-2">RFC 5489</a>.
+	 */
+	private final String identity;
+	private final byte[] pointEncoded;
+	
+	/**
+	 * Constructor {@link EcdhPskClientKeyExchange}. The {@link EcdhPskClientKeyExchange} message include 
+	 * an psk identity as clear text for server to identify which psk key is used, 
+	 * and the ephemeral public key of the client.  
+	 * @param hint - PSK identity as clear text
+	 * @param clientPublicKey - ephemeral public key of client
+	 * @param peerAddress
+	 */
+	public EcdhPskClientKeyExchange(String hint, PublicKey clientPublicKey, InetSocketAddress peerAddress) {
+		super(peerAddress);
+		this.identity = hint;
+		this.identityEncoded = hint.getBytes(UTF_8);
+		ECPublicKey publicKey = (ECPublicKey) clientPublicKey;
+		ECPoint point = publicKey.getW();
+		ECParameterSpec params = publicKey.getParams();
+			
+		this.pointEncoded = ECDHECryptography.encodePoint(point, params.getCurve());
+	}
+	
+	/**
+	 * Constructor {@link EcdhPskClientKeyExchange}. 
+	 * @param hintEncoded - opaque encoded PSK identity hint for server
+	 * @param pointEncoded - ephemeral public key of client (encoded point) 
+	 * @param peerAddress
+	 */
+	public EcdhPskClientKeyExchange(byte[] hintEncoded, byte[] pointEncoded, InetSocketAddress peerAddress) {
+		super(peerAddress);
+		this.identityEncoded = Arrays.copyOf(hintEncoded, hintEncoded.length);
+		this.identity = new String(this.identityEncoded,UTF_8);
+		this.pointEncoded = Arrays.copyOf(pointEncoded, pointEncoded.length);
+	}
+
+	@Override
+	public byte[] fragmentToByteArray() {
+		DatagramWriter writer = new DatagramWriter();
+		writer.write(identityEncoded.length, IDENTITY_LENGTH_BITS);
+		writer.writeBytes(identityEncoded);
+		writer.write(pointEncoded.length, LENGTH_BITS);
+		writer.writeBytes(pointEncoded);
+		return writer.toByteArray();
+	}
+	
+	/**
+	 * This method returns the {@link EcdhPskClientKeyExchange} message from the byte array.
+	 * @param byteArray
+	 * @param peerAddress
+	 * @return
```

Comment:
you could document here that the returned message is a `{@code EcdhPskClientKeyExchange}` ...